pkg/certs_test.go

```diff
@@ -0,0 +1,53 @@
+package courier_test
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/trisacrypto/courier/pkg/api/v1"
+)
+
+func (s *courierTestSuite) TestStoreCertificatePassword() {
+	require := s.Require()
+
+	s.Run("HappyPath", func() {
+		// Configure the store mock to return a successful response
+		req := &api.StorePasswordRequest{
+			ID:       "certID",
+			Password: "password",
+		}
+		s.store.OnUpdatePassword = func(name string, password []byte) error {
+			require.Equal(req.ID, name, "wrong password name passed to store")
+			require.Equal([]byte(req.Password), password, "wrong password passed to store")
+			return nil
+		}
+		defer s.store.Reset()
+
+		// Make a request to the endpoint
+		err := s.client.StoreCertificatePassword(context.Background(), req)
+		require.NoError(err, "could not store certificate password")
+	})
+
+	s.Run("MissingPassword", func() {
```

Comment:
how about also adding a missing ID check?

Reply:
It's hard to test that on the server side since the ID is part of the URL and it would be a 404, however I've added a client test to ensure that the client catches that case, thanks for the suggestion!